### PR TITLE
Fix duplicate record type definition generation and improve code coverage

### PR DESCRIPTION
**Pull request by tharindulak**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by tharindu-nw — APPROVED** [review]

**Review by tharindu-nw — APPROVED** [review]

**Review by madushajg — DISMISSED** [review]

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java [hunk 1]
```diff
@@ -124,7 +125,7 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
         } else {
             // Sets generated type definition code block
             NodeList<ModuleMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createNodeList(
-                    typeDefinitionNodeList.toArray(new TypeDefinitionNode[0]));
+                    typeDefinitionNodes.values().toArray(new TypeDefinitionNode[0]));
```
Comment (nadeeshaan) [inline misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java:128]: Any specific reason to use the `toArray`? Can't we use stream API and map to a list and provide that to the `createNodeList`?

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java [hunk 2]
```diff
@@ -139,22 +140,22 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
...
-            return new ArrayList<>(typeDefinitionNodeList);
+            return new LinkedHashMap<>(typeDefinitionNodes);
         }
 
         Components components = openApi.getComponents();
 
         if (components.getSchemas() == null) {
-            return new ArrayList<>(typeDefinitionNodeList);
+            return new LinkedHashMap<>(typeDefinitionNodes);
         }
```
Comment (nadeeshaan) [inline misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java:159]: You can merge these two if clauses together

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java [hunk 3]
```diff
@@ -179,7 +180,7 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
                 Map<String, Schema> fields = schema.getValue().getProperties();
```
Comment (nadeeshaan) [inline misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java:180]: There is a raw usage of the paramterized type error. Shall we fix that?

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java [hunk 4]
```diff
@@ -191,12 +192,12 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
                         NodeFactory.createRecordTypeDescriptorNode(recordKeyWord, bodyStartDelimiter,
                                 fieldNodes, null, bodyEndDelimiter);
                 if (isRecordTypeDescriptor) {
-                    typeDefinitionNodeList.add(recordTypeDescriptorNode);
+                    typeDefinitionNodes.put(schema.getKey().trim(), recordTypeDescriptorNode);
```
Comment (nadeeshaan) [inline misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java:195]: Extract the `schema.getKey().trim()` expression to a separate variable and reuse is

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java (hunk 3 above)
Comment (tharindulak) [reply misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java:180]: I did get what you mean in this comment. Could please explain a bit more about what I have to do here?

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java (hunk 4 above)
Comment (tharindulak) [reply misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java]: Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34525/commits/9129a28d18a2ad3bae9e19515f274b8ae939c033

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java (hunk 2 above)
Comment (tharindulak) [reply misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java]: Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34525/commits/9129a28d18a2ad3bae9e19515f274b8ae939c033

File: misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java (hunk 1 above)
Comment (tharindulak) [reply misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java]: Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34525/commits/9129a28d18a2ad3bae9e19515f274b8ae939c033

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**